inference/worker/chat_chain_utils.py

```diff
@@ -153,28 +152,39 @@ def prepare_json(json_str: str) -> str:
 
 Here is the fixed JSON object string:</s>{V2_ASST_PREFIX}"""
             logger.warning(f"JSON Fix Prompt: {prompt}")
-            out = llm_parser.generate(prompts=[prompt]).generations[0][0].text
+            out = llm_json_parser.generate(prompts=[prompt]).generations[0][0].text
             out = out[: out.find("}") + 1]
             logger.warning(f"JSON Fix Output: {out}")
             return out
 
     return fixed_json
 
 
-def use_tool(tool_name: str, tool_input: str, tools: list) -> str:
-    best_match, best_similarity = max(
-        ((tool, similarity(tool.name, tool_name)) for tool in tools), key=lambda x: x[1], default=(None, 0)
+def select_tool(tool_name: str, tools: list[Tool]) -> Tool | None:
+    tool = next((t for t in tools if t.name in tool_name), None)
+    if tool:
+        return tool
+    tool, tool_similarity = max(
+        ((t, similarity(t.name, tool_name)) for t in tools),
+        key=lambda x: x[1],
+        default=(None, 0),
     )
```

Comment:
I would leave this comment somewhere if possible, just so we do not have a better model in the future, and similarity like it's still used, etc...
# This should become stricter and stricter as we get better models

Reply:
Ok, added